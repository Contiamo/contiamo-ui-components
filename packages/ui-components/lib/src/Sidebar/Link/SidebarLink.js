"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var react_router_dom_1 = require("react-router-dom");
var glamorous_1 = require("glamorous");
var contiamo_ui_utils_1 = require("contiamo-ui-utils");
var style = function (_a) {
    var theme = _a.theme, color = _a.color, disabled = _a.disabled, active = _a.active;
    var backgroundColor = color ? contiamo_ui_utils_1.hexOrColor(color)(theme.colors.palette && theme.colors.palette[color]) : "#fff", textColor = contiamo_ui_utils_1.readableTextColor(backgroundColor)([theme.colors.palette.grey80, "white"]), disabledStyle = disabled ? { opacity: 0.25 } : { opacity: 1 };
    return __assign({ backgroundColor: backgroundColor }, theme.typography.body, { position: "relative", display: "flex", padding: theme.spacing / 1.5 + "px " + theme.spacing + "px", transition: "background-color .1s ease", cursor: "pointer", 
        // react-router <Link /> wraps an <a> which can be underlined by default so
        textDecoration: "none", color: textColor }, disabledStyle, { "&:link, &:visited": {
            color: textColor
        }, ":focus": {
            outline: 0,
            backgroundColor: contiamo_ui_utils_1.darken(backgroundColor)(10)
        }, "&.co_link + .co_link": {
            borderTop: "1px solid",
            borderColor: theme.colors.usage.subContentSeparatorLine
        }, ":hover": {
            backgroundColor: contiamo_ui_utils_1.darken(backgroundColor)(2),
            // The text color needs to change too if it gets too dark üòÅ
            // Also, here's a prime benefit of functional JS: function composition!
            color: contiamo_ui_utils_1.readableTextColor(contiamo_ui_utils_1.darken(backgroundColor)(5))(["black", "white"])
        } });
};
exports.style = style;
var Symbol = glamorous_1.default.div({
    marginLeft: "auto"
});
var SidebarLink = function (_a) {
    var className = _a.className, children = _a.children, to = _a.to, onClick = _a.onClick, symbol = _a.symbol;
    // if this is expected to work with react-router,
    if (to) {
        return (React.createElement(react_router_dom_1.Link, { to: to ? to : "", className: className + " co_link" },
            children,
            symbol ? React.createElement(Symbol, null, symbol) : ""));
    }
    return (React.createElement("div", { onClick: onClick, className: className + " co_link" },
        children,
        symbol ? React.createElement(Symbol, null, symbol) : ""));
};
exports.SidebarLink = SidebarLink;
exports.default = glamorous_1.default(SidebarLink)(style);
//# sourceMappingURL=SidebarLink.js.map