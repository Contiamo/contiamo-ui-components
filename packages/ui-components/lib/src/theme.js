"use strict";
// Type definitions
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
// Default theme definition
var palette = {
    info: "#1499CE",
    success: "#00b34d",
    warning: "#FFAE00",
    error: "#DE1A1A",
    white: "#FFFFFF",
    black: "#000000",
    grey10: "#F5F5F5",
    grey20: "#F1F1F1",
    grey30: "#D0D9E5",
    grey40: "#C6D1E1",
    grey50: "#BBCADC",
    grey60: "#999999",
    grey70: "#8092B0",
    grey80: "#747474",
    grey90: "#445873"
};
var usageColors = {
    bodyText: "#555f61",
    lightText: "#969696",
    emphasizedText: "#373d3f",
    contentBorder: "#dadada",
    bodyBackground: "#F1F1F1",
    cardBackground: palette.white,
    contentSeparatorLine: "#f2f2f2",
    subContentSeparatorLine: "#f8f8f8"
};
var colors = {
    palette: palette,
    usage: usageColors
};
var baseTypography = {
    lineHeight: "1.5",
    textTransform: "none",
    letterSpacing: "normal"
};
var typography = {
    title: __assign({}, baseTypography, { fontSize: 22, fontWeight: 600 }),
    heading1: __assign({}, baseTypography, { fontSize: 13, fontWeight: 700, color: usageColors.emphasizedText }),
    heading2: __assign({}, baseTypography, { fontSize: 13, fontWeight: 600, textTransform: "uppercase", color: usageColors.lightText, "&::before": {
            content: "Â» "
        } }),
    body: __assign({}, baseTypography, { fontSize: 13, fontWeight: 400 }),
    small: __assign({}, baseTypography, { fontSize: 12, fontWeight: 400 })
};
var shadows = {
    pressed: "inset 0 1px 1px rgba(0,0,0,0.15)",
    card: "0px 1px 2px #d3d1d1"
};
var defaultTheme = {
    typography: typography,
    shadows: shadows,
    colors: colors,
    spacing: 12,
    baseZIndex: 0
};
exports.default = defaultTheme;
//# sourceMappingURL=theme.js.map