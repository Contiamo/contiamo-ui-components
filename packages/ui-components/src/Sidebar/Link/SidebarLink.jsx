// @flow
import React from "react"
import type { Node } from "react"
import { Link } from "react-router-dom"
import type { LocationShape } from "react-router-dom"
import glamorous, { Div } from "glamorous"

import { hexOrColor, readableTextColor, darken } from "contiamo-ui-utils"

import withTooltip from "../../Tooltip/withTooltip"

const SidebarLink = ({
    className,
    children,
    to,
    onClick,
    symbol
  }: {
    className: string,
    children: Node,
    to?: string | LocationShape,
    onClick?: void,
    symbol?: string,
  }) => {
    let Component = Div // By default, use a standard styled div.

    // if this is expected to work with react-router,
    if (to) {
      Component = Link // use a <Link /> since it supports props.to.
    }

    return (
      <Component to={to ? to : ""} onClick={onClick} className={`${className} SideNavigationLink`}>
        {children}
        {symbol
          ? <div className="symbol">
            {symbol}
          </div>
          : ""}
      </Component>
    )
  },
  style = ({ theme, color, disabled }: { theme: THEME, color?: string, disabled?: boolean }) => {
    const backgroundColor = color ? hexOrColor(color)(theme.colors && theme.colors[color]) : "#fff",
      textColor = readableTextColor(backgroundColor)([theme.greys ? theme.greys["80"] : "#747474", "white"]),
      disabledStyle = disabled ? { opacity: 0.25 } : { opacity: 1 }

    return {
      position: "relative",
      display: "flex",
      padding: theme.spacing >= 0 ? theme.spacing / 2 : 8,
      transition: "background-color .1s ease",
      cursor: "pointer",
      fontSize: ".9rem",
      // react-router <Link /> wraps an <a> which can be underlined by default so
      textDecoration: "none",
      color: textColor,
      backgroundColor,
      ...disabledStyle,

      "&:link, &:visited": {
        color: textColor
      },

      "&.SideNavigationLink + .SideNavigationLink": {
        borderTop: "1px solid",
        borderColor: theme.greys ? theme.greys["10"] : "#F5F5F5"
      },

      ":hover": {
        backgroundColor: darken(backgroundColor)(5),

        // The text color needs to change too if it gets too dark 😁
        // Also, here's a prime benefit of functional JS: function composition!
        color: readableTextColor(darken(backgroundColor)(5))(["black", "white"])
      },
      // Symbol goes on the right.
      "& > .symbol": {
        marginLeft: "auto"
      }
    }
  }
export default glamorous(withTooltip(SidebarLink))(style)
export { SidebarLink, style }
