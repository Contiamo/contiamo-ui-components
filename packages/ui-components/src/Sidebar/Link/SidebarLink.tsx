import * as React from "react"
import { SFC } from "react"
import { Link } from "react-router-dom"

import glamorous, { Div } from "glamorous"

import { hexOrColor, readableTextColor, darken } from "contiamo-ui-utils"

type Props = {
  className?: string
  children: React.ReactNode
  to?: string
  onClick?: () => void
  symbol?: string
  theme?: Theme
  color?: string
  disabled?: boolean
  tooltip?: string
  active?: boolean
}

const style: {} = ({ theme, color, disabled, active }: Props) => {
  const backgroundColor = color ? hexOrColor(color)(theme.colors.palette && theme.colors.palette[color]) : "#fff",
    textColor = readableTextColor(backgroundColor)([theme.colors.palette.grey80, "white"]),
    disabledStyle = disabled ? { opacity: 0.25 } : { opacity: 1 }

  return {
    position: "relative",
    display: "flex",
    padding: `${theme.spacing / 1.5}px ${theme.spacing}px`,
    transition: "background-color .1s ease",
    cursor: "pointer",
    // react-router <Link /> wraps an <a> which can be underlined by default so
    textDecoration: "none",
    color: active ? theme.colors.palette.primary : textColor,
    backgroundColor,
    ...disabledStyle,

    "&:link, &:visited": {
      color: textColor,
    },

    "&.SideNavigationLink + .SideNavigationLink": {
      borderTop: "1px solid",
      borderColor: theme.colors.usage.subContentSeparatorLine,
    },

    ":hover": {
      backgroundColor: darken(backgroundColor)(2),

      // The text color needs to change too if it gets too dark 😁
      // Also, here's a prime benefit of functional JS: function composition!
      color: readableTextColor(darken(backgroundColor)(5))(["black", "white"]),
    },
    // Symbol goes on the right.
    "& > .symbol": {
      marginLeft: "auto",
    },
  }
}

const SidebarLink: SFC<Props> = ({ className, children, to, onClick, symbol }: Props) => {
  // if this is expected to work with react-router,
  if (to) {
    return (
      <Link to={to ? to : ""} className={`${className} SideNavigationLink`}>
        {children}
        {symbol ? <div className="symbol">{symbol}</div> : ""}
      </Link>
    )
  }

  return (
    <div onClick={onClick} className={`${className} SideNavigationLink`}>
      {children}
      {symbol ? <div className="symbol">{symbol}</div> : ""}
    </div>
  )
}
export default glamorous(SidebarLink)(style)
export { SidebarLink, style }
