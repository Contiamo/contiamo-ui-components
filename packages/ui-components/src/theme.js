// @flow
const THEME_COLORS: COLORS = {
  primary: "#22205F",
  accent: "#3333FF",
  secondary: "#00CB98",
  tertiary: "#FFFF98",
  warn: "#DE1A1A"
}

const THEME_GREYS: GREYS = {
  "10": "#F5F5F5",
  "20": "#F1F1F1",
  "30": "#D0D9E5",
  "40": "#C6D1E1",
  "50": "#BBCADC",
  "60": "#999999",
  "70": "#8092B0",
  "80": "#747474",
  "90": "#445873",
  "100": "#2D3842",
  white: "#FFFFFF"
}

const DEFAULT_THEME: THEME = {
  colors: THEME_COLORS,
  greys: THEME_GREYS,
  fonts: {
    fontFamily: "Proxima Nova",
    fontSize: 13,
    WebkitFontSmoothing: "subpixel-antialiased",
    textRendering: "optimizeLegibility"
  },
  spacing: 12,
  baseZIndex: 0
}

export default DEFAULT_THEME
