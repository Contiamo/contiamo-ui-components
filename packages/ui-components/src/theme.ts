const palette: Palette = {
  info: "#1499CE",
  success: "#00b34d",
  warning: "#FFAE00",
  error: "#DE1A1A",
  white: "#FFFFFF",
  black: "#000000",
  grey10: "#F5F5F5",
  grey20: "#F1F1F1",
  grey30: "#D0D9E5",
  grey40: "#C6D1E1",
  grey50: "#BBCADC",
  grey60: "#999999",
  grey70: "#8092B0",
  grey80: "#747474",
  grey90: "#445873",
}

const usageColors: UsageColors = {
  bodyText: "#555f61",
  lightText: "#969696",
  emphasizedText: "#373d3f",
  contentBorder: "#dadada",
  bodyBackground: palette.grey20,
  cardBackground: palette.white,
  contentSeparatorLine: "#f2f2f2",
  subContentSeparatorLine: "#f8f8f8",
}

const colors: ThemeColors = {
  palette,
  usage: usageColors
}

const baseTypography: { lineHeight: string; textTransform: string; letterSpacing: number | "normal" } = {
  lineHeight: "1.5",
  textTransform: "none",
  letterSpacing: "normal",
}

const typography: ThemeTypography = {
  title: {
    ...baseTypography,
    fontSize: 22,
    fontWeight: 600,
  },
  heading1: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 700,
    color: usageColors.emphasizedText,
  },
  heading2: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 600,
    textTransform: "uppercase",
    color: usageColors.lightText,
    "&::before": {
      content: "Â» ",
    },
  },
  body: {
    ...baseTypography,
    fontSize: 13,
    fontWeight: 400,
  },
  small: {
    ...baseTypography,
    fontSize: 12,
    fontWeight: 400,
  },
}

const shadows = {
  pressed: "inset 0 1px 1px rgba(0,0,0,0.15)",
  card: "0px 1px 2px #d3d1d1",
}

const defaultTheme: Theme = {
  typography,
  shadows,
  colors,
  spacing: 12,
  baseZIndex: 0,
}

export default defaultTheme
