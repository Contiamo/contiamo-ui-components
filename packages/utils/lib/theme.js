"use strict";
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var glamorous_1 = require("glamorous");
// Wrap each ui component in its own theme provider to make sure the default
// Contiamo theme is always available. Props are passed along unaltered.
exports.wrapTheme = function (theme) { return function (Comp) {
    return function (props) { return (React.createElement(glamorous_1.ThemeProvider, { theme: theme },
        React.createElement(Comp, __assign({}, props)))); };
}; };
//# sourceMappingURL=theme.js.map