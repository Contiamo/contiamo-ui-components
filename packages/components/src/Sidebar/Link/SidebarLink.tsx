import * as React from "react"
import { Link } from "react-router-dom"
import glamorous, { Div, GlamorousComponent } from "glamorous"
import { Theme } from "@operational/theme"
import { hexOrColor, readableTextColor, darken } from "@operational/utils"

export interface IProps {
  id?: string | number
  style?: {}
  className?: string
  children?: React.ReactNode
  to?: string
  onClick?: () => void
  symbol?: string
  theme?: Theme
  color?: string
  disabled?: boolean
  tooltip?: string
  active?: boolean
}

const style = ({ theme, color, disabled, active }: IProps): {} => {
  const backgroundColor = color ? hexOrColor(color)(theme.colors.palette && theme.colors.palette[color]) : "#fff"
  const textColor = active
    ? theme.colors.usage.link
    : readableTextColor(backgroundColor)([theme.colors.palette.grey80, "white"])
  const disabledStyle = disabled ? { opacity: 0.25 } : { opacity: 1 }

  return {
    backgroundColor,
    ...theme.typography.body,
    fontWeight: active ? 600 : 400,
    position: "relative",
    display: "flex",
    padding: `${theme.spacing / 3}px ${theme.spacing}px`,
    cursor: "pointer",
    // react-router <Link /> wraps an <a> which can be underlined by default so
    textDecoration: "none",
    color: textColor,
    ...disabledStyle,

    "&:link, &:visited": {
      color: textColor
    },

    ":focus": {
      outline: 0
    },

    "&.co_link + .co_link": {
      borderTop: "1px solid",
      borderColor: theme.colors.usage.subContentSeparatorLine
    },

    ":hover": {
      backgroundColor: darken(backgroundColor)(2),

      // The text color needs to change too if it gets too dark üòÅ
      // Also, here's a prime benefit of functional JS: function composition!
      color: active ? textColor : readableTextColor(darken(backgroundColor)(5))(["black", "white"])
    }
  }
}

const Symbol = glamorous.div({
  marginLeft: "auto"
})

const SidebarLink: React.SFC<IProps> = ({ id, style, className, children, to, onClick, symbol }: IProps) => {
  // if this is expected to work with react-router,
  if (to) {
    return (
      <Link to={to ? to : ""} style={style} key={id} className={`${className} co_link`}>
        {children}
        {symbol ? <Symbol>{symbol}</Symbol> : ""}
      </Link>
    )
  }

  return (
    <div key={id} style={style} onClick={onClick} className={`${className} co_link`}>
      {children}
      {symbol ? <Symbol>{symbol}</Symbol> : ""}
    </div>
  )
}
export default glamorous(SidebarLink)(style)
export { SidebarLink, style }
