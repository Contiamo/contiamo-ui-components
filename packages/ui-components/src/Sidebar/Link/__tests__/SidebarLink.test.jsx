// @flow
import React from "react"
import { shallow } from "enzyme"

import { SidebarLink, style } from "../SidebarLink"
import theme from "../../../theme"

describe("SidebarLink", () => {
  it("Should render correctly", () => {
    const renderedComponent = shallow(<SidebarLink className="hi">sup</SidebarLink>)
    expect(renderedComponent).toMatchSnapshot()
  })
  it("Should render a react-router link if given a `to` prop", () => {
    const renderedComponent = shallow(
      <SidebarLink className="hi" to="/route">
        sup
      </SidebarLink>
    )
    expect(renderedComponent).toMatchSnapshot()
  })
  it("Should receive proper styles", () => {
    expect(style({ theme })).toMatchObject({})
  })
  it("Should display a symbol to the right", () => {
    const renderedComponent = shallow(
      <SidebarLink className="hi" symbol="% s  ">
        sup
      </SidebarLink>
    )
    expect(renderedComponent).toMatchSnapshot()
  })
})
