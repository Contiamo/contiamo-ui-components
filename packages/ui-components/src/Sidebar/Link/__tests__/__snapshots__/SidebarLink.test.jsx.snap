// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarLink Should display a symbol to the right 1`] = `
<glamorous.Div
  className="undefined SideNavigationLink"
>
  sup
  <div
    className="symbol"
  >
    % s  
  </div>
</glamorous.Div>
`;

exports[`SidebarLink Should render a react-router link if given a \`to\` prop 1`] = `
<Link
  className="undefined SideNavigationLink"
  replace={false}
  to="/route"
>
  sup
</Link>
`;

exports[`SidebarLink Should render correctly 1`] = `
<glamorous.Div
  className="undefined SideNavigationLink"
>
  sup
</glamorous.Div>
`;
